### ISSUE TITLE
Level editor (v.0.71) fails to compile on Open Pandora (GNU/Linux ARM)

### ISSUE THREAD

**Issue by sviscapi** [davidjoffe/dave_gnukem#63]
Hi David,

As the title says:

/media/Pandora1/dev/dave_gnukem$ make ed
g++ -ffast-math -O3 -pipe -mcpu=cortex-a8 -mfpu=neon -mfloat-abi=softfp -ftree-vectorize -fsingle-precision-constant -fuse-ld=gold -fuse-linker-plugin -fsigned-char -mneon-for-64bits -mvectorize-with-neon-double -Wall -DHAVE_SOUND -DDATA_DIR=/mnt/utmp/davegnukem-magicsam/data -I/mnt/utmp/codeblocks/usr/include/SDL -c src/graph.cpp -o src/graph.o
g++ -ffast-math -O3 -pipe -mcpu=cortex-a8 -mfpu=neon -mfloat-abi=softfp -ftree-vectorize -fsingle-precision-constant -fuse-ld=gold -fuse-linker-plugin -fsigned-char -mneon-for-64bits -mvectorize-with-neon-double -Wall -DHAVE_SOUND -DDATA_DIR=/mnt/utmp/davegnukem-magicsam/data -I/mnt/utmp/codeblocks/usr/include/SDL -c src/ed_standalone_original.cpp -o src/ed_standalone_original.o
In file included from src/ed_standalone_original.cpp:8:0:
src/datadir.h:15:0: warning: "DATA_DIR" redefined
 #define DATA_DIR "data/"
 
<command-line>:0:0: note: this is the location of the previous definition
src/ed_standalone_original.cpp:79:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
 };
 ^
src/ed_standalone_original.cpp:79:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:79:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:79:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:79:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
 };
 ^
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:111:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
 };
 ^
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp:130:1: warning: ISO C++ forbids converting a string constant to ���char*��� [-Wwrite-strings]
src/ed_standalone_original.cpp: In function ���int main(int, char**)���:
src/ed_standalone_original.cpp:274:22: error: too few arguments to function ���void djgFlip(djVisual*, djVisual*, bool)���
    djgFlip(pVis, NULL);
                      ^
In file included from src/djgamelib.h:4:0,
                 from src/ed_standalone_original.cpp:9:
src/djgraph.h:128:19: note: declared here
 extern void       djgFlip( djVisual * pVisDest, djVisual * pVisSrc, bool bScaleView );
                   ^~~~~~~
src/ed_standalone_original.cpp:287:2: error: ���level_init��� was not declared in this scope
  level_init();
  ^~~~~~~~~~
src/ed_standalone_original.cpp:287:2: note: suggested alternative: ���level_set���
  level_init();
  ^~~~~~~~~~
  level_set
src/ed_standalone_original.cpp:292:55: error: invalid conversion from ���const char*��� to ���char*��� [-fpermissive]
   szfilename = g_pCurMission->GetLevel(i)->GetFilename();
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/ed_standalone_original.cpp:515:58: error: invalid conversion from ���const char*��� to ���char*��� [-fpermissive]
      szfilename = g_pCurMission->GetLevel(i)->GetFilename();
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/ed_standalone_original.cpp:603:15: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for ( i=0; i<g_aMessages.size(); i++ )
              ~^~~~~~~~~~~~~~~~~~~
src/ed_standalone_original.cpp:620:21: error: too few arguments to function ���void djgFlip(djVisual*, djVisual*, bool)���
   djgFlip(pVis, NULL);
                     ^
In file included from src/djgamelib.h:4:0,
                 from src/ed_standalone_original.cpp:9:
src/djgraph.h:128:19: note: declared here
 extern void       djgFlip( djVisual * pVisDest, djVisual * pVisSrc, bool bScaleView );
                   ^~~~~~~
src/ed_standalone_original.cpp:625:2: error: ���level_done��� was not declared in this scope
  level_done();
  ^~~~~~~~~~
src/ed_standalone_original.cpp:625:2: note: suggested alternative: ���level_save���
  level_done();
  ^~~~~~~~~~
  level_save
src/ed_standalone_original.cpp: In function ���void sprite_show_extra(int, int)���:
src/ed_standalone_original.cpp:973:2: warning: this ���else��� clause does not guard... [-Wmisleading-indentation]
  else
  ^~~~
src/ed_standalone_original.cpp:978:3: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ���else���
   DrawString(
   ^~~~~~~~~~
src/ed_standalone_original.cpp: In function ���void level_draw_levelname()���:
src/ed_standalone_original.cpp:1269:61: error: invalid conversion from ���const char*��� to ���char*��� [-fpermissive]
  szfilename = g_pCurMission->GetLevel(g_nLevel)->GetFilename();
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
Makefile:123: recipe for target 'src/ed_standalone_original.o' failed
make: *** [src/ed_standalone_original.o] Error 1
/media/Pandora1/dev/dave_gnukem$

Cheers,

Sam a.k.a "Magic Sam" from the Open Pandora community

P.S: I'll reply to your e-mail shortly

**Comment by davidjoffe** [comment]
Was it possible to compile the standalone level editor previous, or was it the first time you tried it? Level editor used to be standalone application, someone (I think EMH) integrated it into the game while I had abandoned it, and since I started again on it I literally hadn't even tried to 'make ed' - I was working on the assumption (possibly incorrect) that that was no longer relevant because of the integrated level editor. If it is was working, then I can resurrect these makefile items and try get it working again, though I feel it's probably not necessary (and maybe not worth the effort to maintain) BOTH an integrated level editor and a standalone editor, not sure if you have any thoughts on that from OpenPandora perspective.
